### PR TITLE
[#145964569] Tokenized public dashboard

### PR DESCRIPTION
**Pull request by waggles**
Depends on #1067 , see https://github.com/tablexi/nucore-open/compare/dashboard-145965563...public-dashboard-145964569
Phase 1 of https://www.pivotaltracker.com/story/show/145964569

Does include:
* Adds new layout and route to be used by tablet
* Add token to identify `SecureRoom`s

Does not include yet:
* Auto-refresh/show last refreshed at
* Specifically accurate styling of the partial

### CODE REVIEW

File: vendor/engines/secure_rooms/db/migrate/20170525143813_add_token_to_secure_room.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+  class Product < ActiveRecord::Base
+  end
+
+  def change
+    add_column :products, :dashboard_token, :string
+    add_index :products, :dashboard_token
+
+    Product.reset_column_information
+    Product.where(type: :SecureRoom).find_each { |room| room.update_column(:dashboard_token, SecureRandom.uuid) }
```
Comment (jhanggi) [inline vendor/engines/secure_rooms/db/migrate/20170525143813_add_token_to_secure_room.rb]: It's a little awkward to use a symbol for `:SecureRoom`. I'd probably just use a string.

Comment (jhanggi) [inline vendor/engines/secure_rooms/db/migrate/20170525143813_add_token_to_secure_room.rb]: And will this `db:rollback` properly? I always forget when you run some extra code.

File: app/views/layouts/plain.html.haml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE html>
+%html
+  %head
+    = render :partial => '/shared/head'
+  %body
+    #content
+      .container
+        .row
+          .span12
+            = render partial: 'shared/flashes'
```
Comment (jhanggi) [inline app/views/layouts/plain.html.haml]: Double quotes and no `partial`.

File: app/views/layouts/plain.html.haml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE html>
+%html
+  %head
+    = render :partial => '/shared/head'
```
Comment (jhanggi) [inline app/views/layouts/plain.html.haml]: Double quotes and no partial.

File: vendor/engines/secure_rooms/app/views/secure_rooms/shared/_dashboard_with_header.html.haml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+%h3= room.name
+= link_to "Link", facility_secure_room_occupancies_path(room.facility, room.dashboard_token)
```
Comment (jhanggi) [inline vendor/engines/secure_rooms/app/views/secure_rooms/shared/_dashboard_with_header.html.haml]: I18n

File: vendor/engines/secure_rooms/app/controllers/secure_rooms/secure_room_occupancies_controller.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+module SecureRooms
+
+  class SecureRoomOccupanciesController < ApplicationController
```
Comment (jhanggi) [inline vendor/engines/secure_rooms/app/controllers/secure_rooms/secure_room_occupancies_controller.rb]: It's not a big deal, but `SecureRooms::SecureRoomOccupanciesController` feels a little redundant. What was the main reason for naming it this?

File: vendor/engines/secure_rooms/db/migrate/20170525143813_add_token_to_secure_room.rb (hunk 1 above)
Comment (waggles) [inline vendor/engines/secure_rooms/db/migrate/20170525143813_add_token_to_secure_room.rb]: it seems like it did, just ran again locally

File: app/views/layouts/plain.html.haml (hunk 2 above)
Comment (waggles) [inline app/views/layouts/plain.html.haml]: oh also do you think we want the flashes in this layout?

File: vendor/engines/secure_rooms/app/views/secure_rooms/shared/_dashboard_with_header.html.haml (hunk 4 above)
Comment (waggles) [inline vendor/engines/secure_rooms/app/views/secure_rooms/shared/_dashboard_with_header.html.haml]: should I add "Link" to the `shared` group in the base app or in the engine?

File: vendor/engines/secure_rooms/app/controllers/secure_rooms/secure_room_occupancies_controller.rb (hunk 5 above)
Comment (waggles) [inline vendor/engines/secure_rooms/app/controllers/secure_rooms/secure_room_occupancies_controller.rb]: I don't remember if I had a real reason, but now I prefer it the other way, will change

**Comment by jhanggi** [comment]
@waggles can you clean this PR up now that #1067 is in?

**Comment by waggles** [comment]
merged~

**Comment by jhanggi** [comment]
Looks like something's up with the headers. On the facility view, we've lost the product names.
screen shot 2017-05-26 at 4 32 58 pm

And where do I find the public URL?

**Comment by waggles** [comment]
grar merged over my own toes, is using wrong partial. one sec

File: config/locales/en.yml [hunk 6]
```diff
@@ -96,6 +96,7 @@ en:
     delete: Delete
     dispute: Dispute
     edit: Edit
+    link: Link
```
Comment (jhanggi) [inline config/locales/en.yml]: Can we call this "Public Link" to make it a little clearer what it is? That might mean it no longer makes sense to have this at the top-level `shared`.

**Pull request merged**